lib/puppet/parser/functions/redisget.rb

```diff
@@ -3,20 +3,40 @@
 module Puppet::Parser::Functions
   newfunction(:redisget, :type => :rvalue, :doc => <<-EOS
     Returns the value of the key being looked up or nil if the key does not
-    exist. Takes two arguments, the first being a string value of the key to be
-    looked up and the second is the URL to the Redis service.
+    exist. Takes two arguments with an optional third. The first being a string
+    value of the key to be looked up, the second is the URL to the Redis service
+    and the third optional argument is a default value to be used if the lookup
+    fails.
     EOS
   ) do |args|
 
-    raise(Puppet::ParseError, "redisget(): Wrong number of arguments given (#{args.size} for 2)") if args.size != 2
+    raise(Puppet::ParseError, "redisget(): Wrong number of arguments given (#{args.size} for 2 or 3)") if args.size != 2 and args.size != 3
 
     key = args[0]
     url = args[1]
 
-    raise(Puppet::ParseError, "redisget(): Wrong argument type given (#{key.class} for String") if key.is_a?(String) == false
-    raise(Puppet::ParseError, "redisget(): Wrong argument type given (#{url.class} for String") if url.is_a?(String) == false
+    if args.size == 3
+      default = args[2]
+      raise(Puppet::ParseError, "redisget(): Wrong argument type given (#{default.class} for String) for arg3 (default)") if default.is_a?(String) == false
+    end
 
-    redis = Redis.new(:url => url)
-    redis.get(key)
+    raise(Puppet::ParseError, "redisget(): Wrong argument type given (#{key.class} for String) for arg1 (key)") if key.is_a?(String) == false
+    raise(Puppet::ParseError, "redisget(): Wrong argument type given (#{url.class} for String) for arg2 (url)") if url.is_a?(String) == false
+
+    begin
```

Comment:
changed the variable name and added the exception message to the output as requested